x-pack/elastic-agent/pkg/composable/providers/kubernetes/kubernetes.go

```diff
@@ -20,6 +21,10 @@ const (
 	PodPriority = 0
 	// ContainerPriority is the priority that container mappings are added to the provider.
 	ContainerPriority = 1
+	// NodePriority is the priority that node mappings are added to the provider.
+	NodePriority = 0
+	// ServicePriority is the priority that service mappings are added to the provider.
+	ServicePriority = 0
```

Comment:
These should each be unique and they should be tiered. I think the proper order here should be:

```
NodePriority = 0
PodPriority = 1
ContainerPriority = 2
ServicePriority = 3
```

That is because a Node contains a pod, a pod contains containers, and services point at pods.